core/vm/instructions.go

```diff
@@ -569,6 +573,11 @@ func opCreate(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]b
 		input        = scope.Memory.GetCopy(int64(offset.Uint64()), int64(size.Uint64()))
 		gas          = scope.Contract.Gas
 	)
+
+	if interpreter.readOnly {
+		return nil, ErrWriteProtection
+	}
+
```

Comment:
I'd move this check up to before the `var` clause, it feels more correct to check errors before we start manipulating the stack.

Reply:
Good point, did it on all places.